// @metlife-apac/input/src/input.stories.js
import React from "react";
import { Input } from "./index";

export default {
  title: "MetLife APAC UI Library/Input",
  component: Input,
  argTypes: {},
};

const Template = (args) => (
  <div className="login">
    <form className="login__container" onSubmit={() => {}}>
      <Input {...args} />
    </form>
  </div>
);

export const InputOutlined = Template.bind({});
InputOutlined.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "",
  inputStyle: 'outlined',
};

export const InputOutlinedShowhideIcon = Template.bind({});
InputOutlinedShowhideIcon.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "",
  inputStyle: 'outlined',
  iconType: 'showhide'
};

export const InputOutlinedInfoIcon = Template.bind({});
InputOutlinedInfoIcon.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "",
  inputStyle: 'outlined',
  iconType: 'info',
  infoContent: 'Info sample content'
};

export const InputOutlinedToolTip = Template.bind({});
InputOutlinedToolTip.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "",
  inputStyle: 'outlined',
  toolTipTitle: 'What is this?',
  toolTipContent: 'This is sample tooltip content'
};

export const InputOutlinedError = Template.bind({});
InputOutlinedError.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "Error",
  inputStyle: 'underlined',
}

export const InputUnderlined = Template.bind({});
InputUnderlined.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "",
  inputStyle: 'underlined',
};

export const InputUnderlinedShowhideIcon = Template.bind({});
InputUnderlinedShowhideIcon.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "",
  inputStyle: 'underlined',
  iconType: 'showhide'
};

export const InputUnderlinedInfoIcon = Template.bind({});
InputUnderlinedInfoIcon.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "",
  inputStyle: 'underlined',
  iconType: 'info',
  infoContent: 'Info sample content'
};

export const InputUnderlinedToolTip = Template.bind({});
InputUnderlinedToolTip.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "",
  inputStyle: 'underlined',
  toolTipTitle: 'What is this?',
  toolTipContent: 'This is sample tooltip content'
};

export const InputUnderlinedError = Template.bind({});
InputUnderlinedError.args = {
  "data-testid": "login-username",
  className: "",
  name: "username",
  label: "Username",
  value: "",
  placeholder: "Username",
  ref: {},
  error: "Error",
  inputStyle: 'underlined',
}
