// @metlife-apac/loader/src/loader.stories.js
import React from "react";
import { Loader } from "./index";

export default {
  title: "MetLife APAC UI Library/Loader",
  component: Loader,
  argTypes: {},
};

const triggerDarkMode = (args) => {
  if (args.isDarkMode) {
    return {
      ...args,
      className: "met-loader__storybook-dark",
      loaderMessageClass: "met-loader__message__storybook-dark",
      loaderSpinnerClass: "met-loader__spinner__storybook-dark",
    };
  } else {
    return {
      ...args,
      className: "met-loader__storybook-light",
      loaderMessageClass: "met-loader__message__storybook-light",
      loaderSpinnerClass: "met-loader__spinner__storybook-light",
    };
  }
};

const Template = (args) => {

  const storybookClass = triggerDarkMode(args);

  return <Loader {...storybookClass} />;
};

export const Primary = Template.bind({});
Primary.args = {
  testId: "",
  isLoading: true,
  className: "met-loader__storybook-light",
  message: "Page is loading...",
  loaderMessageClass: "met-loader__message__storybook-light",
  loaderSpinnerClass: "met-loader__spinner__storybook-light",
  isDarkMode: false,
};
