// @metlife-apac/accordion/src/accordion.stories.js
import React from "react";
import { Accordion } from "./index";
import { List } from "../../list/src/index";

export default {
  title: "MetLife APAC UI Library/Accordion",
  component: Accordion,
  argTypes: {},
};

const agentInfo = [
  {
    id: "branch",
    label: "KL",
    value: "",
  },
  {
    id: "name",
    label: "Terry Chew",
    value: "",
  },
  {
    id: "code",
    label: "GMT6070",
    value: "",
  },
];

const renderListItem = ({ label, value, id }, idx) => {
  return (
    <div className="policydetail__item" data-testid={`${id}-${idx}`}>
      <div className="policydetail__item-label">{label}</div>
      <div className="policydetail__item-value">{value}</div>
    </div>
  );
};

const Template = (args) => (
  <div className="policydetail">
    <div className="policydetail__grouplist">
      <Accordion {...args}>
        <List
          data-testid="agentInfo-list"
          data={agentInfo}
          renderItem={renderListItem}
        />
      </Accordion>
    </div>
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  testId: "agentInfo",
  logo: "",
  logoClass: "",
  titleRightIconClass: "",
  className: "policydetail__accordion",
  title: "Agent Info",
  titleClass: "policydetail__accordianTitle",
  isExpand: false,
  disabled: false,
  isArrow: false,
};

export const PrimaryWithIcon = Template.bind({});
PrimaryWithIcon.args = {
  testId: "agentInfo",
  logo: "",
  logoClass: "",
  titleRightIconClass: "",
  className: "policydetail__accordion",
  title: "Agent Info",
  titleClass: "policydetail__accordianTitle",
  icon: './sample.png',
  isExpand: false,
  disabled: false,
  isArrow: false,
};
