// @metlife-apac/table/src/table.stories.js
import React from 'react';
import { Table } from "./index";
import 'bootstrap/dist/css/bootstrap.min.css';

export default {
    title: 'MetLife APAC UI Library/Table',
    component: Table,
    argTypes: {},
};

const tableData = [
    {
        propspectsName: 'Harry Kane',
        productName: 'Live Life Premier',
        lastUpdated: <div className='test'>12/05/2019</div>,
        status: <a href='#'>Pending Premium</a>,
    },
    {
        propspectsName: 'Casdan Moro',
        productName: 'Live Life Premier',
        lastUpdated: <div className='test'>12/05/2019</div>,
        status: <a href='#'>Pending Premium</a>,
    },
    {
        propspectsName: 'Casdan Moro',
        productName: 'Investor Advantage Premier',
        lastUpdated: <div className='test'>5/01/2019</div>,
        status: <a href='#'>Pending Premium</a>,
    },
    {
        propspectsName: 'Marissa Moore',
        productName: 'Live Life Premier',
        lastUpdated: <div className='test'>6/05/2019</div>,
        status: <a href='#'>Pending Premium</a>,
    },
];

const columnData = [
    {
        label: 'Prospects Name',
        value: 'propspectsName',
    },
    {
        label: 'Product Name',
        value: 'productName',
    },
    {
        label: 'Last Updated',
        value: 'lastUpdated',
    },
    {
        label: 'Status',
        value: 'status',
    },
];

const sortList = [
    {
        label: 'Recent Activity (Newest)',
        value: 'lastUpdated',
    },
];

const Template = (args) => (
    <div className='application'>
        <Table {...args} />
    </div>
);

export const Primary = Template.bind({});
Primary.args = {
    testId: '',
    containerClassName: '',
    className:'application__table',
    dataList:tableData,
    columns:columnData,
    onRowClick:() => {},
    sortList:sortList,
    onSort:() => {},
    hasIndicator: false
};
