// @metlife-apac/button/src/button.stories.js
import React from "react";

import { Button } from "./index";

export default {
  title: "MetLife APAC UI Library/Button",
  component: Button,
  argTypes: {},
};

const Template = (args) => <Button {...args}>{"Button"}</Button>;

export const PrimaryActive = Template.bind({});
PrimaryActive.args = {
  type: 'primary',
  onClick: () => {},
  children: "",
  testId: "",
  className: "",
};

export const PrimaryDarkActive = Template.bind({});
PrimaryDarkActive.args = {
  type: 'primarydark',
  onClick: () => {},
  children: "",
  testId: "",
  className: "",
};

export const SecondaryActive = Template.bind({});
SecondaryActive.args = {
  type: 'secondary',
  onClick: () => {},
  children: "",
  testId: "",
  className: "",
};

export const SecondaryWhiteActive = Template.bind({});
SecondaryWhiteActive.args = {
  type: 'secondarywhite',
  onClick: () => {},
  children: "",
  testId: "",
  className: "",
};

export const SecondaryBlackActive = Template.bind({});
SecondaryBlackActive.args = {
  type: 'secondaryblack',
  onClick: () => {},
  children: "",
  testId: "",
  className: "",
};

export const Disabled = Template.bind({});
Disabled.args = {
  type: 'secondaryblack',
  onClick: () => {},
  children: "",
  testId: "",
  disabled: true,
};
