// @metlife-apac/list/src/list.stories.js
import React from "react";
import { List } from "./index";

export default {
  title: "MetLife APAC UI Library/List",
  component: List,
  argTypes: {},
};

const agentInfo = [
  {
    id: "branch",
    label: "Branch",
    value: "",
  },
  {
    id: "name",
    label: "Name",
    value: "",
  },
  {
    id: "code",
    label: "Code",
    value: "",
  },
];


const agentInfoCollapsed = [
  {
    id: "branch",
    label: "Branch",
    value: "",
  },
  {
    id: "name",
    label: "Name",
    value: "",
  },
  {
    id: "code",
    label: "Code",
    value: "",
  },
  {
    id: "mobile",
    label: "Mobile",
    value: "",
  },
  {
    id: "address",
    label: "Address",
    value: "",
  },
  {
    id: "license",
    label: "License",
    value: "",
  },
];

const renderListItem = ({ label, value, id }, idx) => {
  return (
    <div className="policydetail__item" data-testid={`${id}-${idx}`}>
      <div className="policydetail__item-label">{label}</div>
      <div className="policydetail__item-value">{value}</div>
    </div>
  );
};

const Template = (args) => (
  <List {...args} />
);

export const Default = Template.bind({});
Default.args = {
  "data-testid": "agentInfo-list",
  testId: "",
  className: "",
  collapsedSize: "",
  data: agentInfo,
  renderItem: renderListItem,
};

export const Collapsed = Template.bind({});
Collapsed.args = {
  "data-testid": "agentInfo-list",
  testId: "",
  className: "",
  collapsedSize: "2",
  data: agentInfoCollapsed,
  renderItem: renderListItem,
};
