// @metlife-apac/dropdown/src/dropdown.js
import React, { forwardRef, useCallback, useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { Icon } from '@metlife-apac/icon';
import ArrowDown from '../../assets/images/arrowdown.png';
import './dropdown.scss';

const Dropdown = forwardRef(
    (
        {
            className,
            dropdownHeight,
            items,
            label,
            placeholderValue,
            value,
            error,
            outline,
            underline,
            autoComplete,
            onChange,
            testId,
            inputClass,
            toolTipTitle,
            toolTipContent,
        },
        ref,
    ) => {
        const [selectedValue, setSelectedValue] = useState(value ?? '');
        const [shouldOpen, setshouldOpen] = useState(false);
        const [filteredItems, setFilteredItems] = useState(items);
        const [inputValue, setInputValue] = useState(
            filteredItems.find((item) => item.value === value)?.label ?? '',
        );
        const iconRef = useRef(null);
        const [showToolTipContent, setShowToolTipContent] = useState(false);

        const onItemClick = useCallback(
            (value) => {
                setSelectedValue(value);
                if (onChange) onChange(value);
            },
            [onChange],
        );

        const toggleDropdown = useCallback(() => {
            setshouldOpen((prevState) => !prevState);
        }, []);

        const onTextChange = useCallback((event) => {
            setInputValue(event.target.value);
        }, []);

        const closeDropdown = () => {
            if (inputValue === '') {
                setInputValue(items.find((item) => item.value === selectedValue)?.label ?? '');
            }

            if (shouldOpen) {
                toggleDropdown();
            }
        };

        const shouldIgnore = (event) => {
            const path = event.path ?? event.composedPath?.();

            return (
                !iconRef.current?.contains(path?.[0]) &&
                !path?.[0]?.className?.includes('met-dropdown__list-item')
            );
        };

        useEffect(() => {
            setshouldOpen(false);
        }, [selectedValue]);

        useEffect(() => {
            if (autoComplete) {
                setFilteredItems(
                    items?.filter((item) => item.label.match(new RegExp(inputValue, 'gi'))),
                );
            }
        }, [autoComplete, items, inputValue]);

        useEffect(() => {
            setInputValue(items.find((item) => item.value === selectedValue)?.label ?? '');
        }, [items, selectedValue]);

        useEffect(() => {
            setSelectedValue(value);
        }, [value]);
        return (
            <div ref={ref} data-testid={testId ?? 'dropdown-wrapper'} className={cn('met-dropdown', className)}>
            <div
                
                data-testid='dropdown-container'
                className={cn(
                    {
                        'met-dropdown--outline': outline,
                        'met-dropdown--underline': underline,
                    }
                )}
            >
                {label && (
                    <div
                        data-testid='dropdown-label'
                        className={cn('met-dropdown__label', {
                            'met-dropdown__label--error': error,
                            'met-dropdown__label--filled': !!inputValue,
                        })}
                    >
                        {label}
                    </div>
                )}
                <div className='met-dropdown__container' onClick={toggleDropdown}>
                    <input
                        data-testid='dropdown-input'
                        className={cn(
                            'met-dropdown__input',
                            {
                                'met-dropdown__input--disabled': !autoComplete,
                            },
                            inputClass,
                        )}
                        type='text'
                        value={inputValue}
                        placeholder={placeholderValue}
                        onChange={onTextChange}
                    />
                    <Icon
                        ref={iconRef}
                        testId='dropdown-icon'
                        className={cn('met-dropdown__icon', {
                            'met-dropdown__icon--opened': shouldOpen,
                        })}
                        src={ArrowDown}
                    />
                </div>
                <div
                    data-testid='dropdown-list'
                    className={cn(
                        'met-dropdown__list',
                        {
                            'met-dropdown__list--opened': shouldOpen,
                        },
                        dropdownHeight,
                    )}
                >
                    {items.length > 1 ? (
                        <>
                            {items.map((item, idx) => (
                                <div
                                    key={idx}
                                    className='met-dropdown__list-item'
                                    onClick={() => onItemClick(item.value)}
                                >
                                    {item.label}
                                </div>
                            ))}
                        </>
                    ) : (
                        <div className='met-dropdown__list-item met-dropdown__list-none'>
                            No options found
                        </div>
                    )}
                </div>
                </div>
                {error && (
                    <span data-testid='dropdown-error' className='met-dropdown--error'>
                        {error}
                    </span>
                )}
                {toolTipTitle && (
                    <div
                        data-testid='tooltipTitle'
                        className='met-dropdown--toolTipTitle'
                        onClick={() => setShowToolTipContent((prevState) => !prevState)}
                    >
                        {toolTipTitle}
                    </div>
                )}
                {showToolTipContent && (
                    <div data-testid='tooltipTitle' className='met-dropdown--toolTipContent'>
                        {toolTipContent}
                    </div>
                )}
            </div>
        );
    },
);

Dropdown.propTypes = {
    className: PropTypes.string,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            label: PropTypes.string.isRequired,
            value: PropTypes.string.isRequired,
        }),
    ),
    label: PropTypes.string.isRequired,
    error: PropTypes.string,
    value: PropTypes.string,
    placeholderValue: PropTypes.string,
    outline: PropTypes.bool,
    underline: PropTypes.bool,
    autoComplete: PropTypes.bool,
    onChange: PropTypes.func,
    testId: PropTypes.string,
    inputClass: PropTypes.string,
    toolTipTitle: PropTypes.string,
    toolTipContent: PropTypes.string,
};

Dropdown.defaultProps = {
    outline: true,
    underline: false,
    autoComplete: false,
};

export default Dropdown;
