// @metlife-apac/dropdown/src/dropdown.stories.js
import React from 'react';
import { Dropdown } from './index'
import 'bootstrap/dist/css/bootstrap.min.css';

export default {
    title: 'MetLife APAC UI Library/Dropdown',
    component: Dropdown,
    argTypes: {
    },
};

const Template = (args) => <div className='met-table__sort'><Dropdown {...args} /></div>;

export const Primary = Template.bind({});
Primary.args = {
    items: [
        {
            label: 'English',
            value: 'en',
        },
        {
            label: 'Korean',
            value: 'kr',
        },
    ],
    label: 'Dropdown',
    underline: true,
    outline: true,
    className: 'about__language about__element',
    value: 'en',
    label: 'Language',
    onChange:()=>{},
    error: '',
    placeholderValue: 'Label',
    autoComplete: false,
    testId: 'dropdown-test',
    inputClass: '',
    toolTipTitle: '',
    toolTipContent: '',
};

export const PrimaryToolTips = Template.bind({});
PrimaryToolTips.args = {
    items: [
        {
            label: 'English',
            value: 'en',
        },
        {
            label: 'Korean',
            value: 'kr',
        },
    ],
    label: 'Dropdown',
    underline: true,
    outline: true,
    className: 'about__language about__element',
    value: 'en',
    label: 'Language',
    onChange:()=>{},
    error: '',
    placeholderValue: 'Label',
    autoComplete: false,
    testId: 'dropdown-test',
    inputClass: '',
    toolTipTitle: 'What is this?',
    toolTipContent: 'This is sample tooltip content'
};
